conans/client/cache/cache.py

```diff
@@ -134,6 +133,22 @@ def get_latest_prev(self, ref):
         prevs = self.get_package_revisions(ref, True)
         return prevs[0] if prevs else None
 
+    def get_latest_pkg_layout(self, ref: ConanReference) -> PackageLayout:
+        """Get the latest PackageLayout given a file reference"""
+        latest_rrev = self.get_latest_rrev(ref)
+        # Given the latest recipe revision we can get the latest package ID and
+        # the corresponding PackageLayout
+        all_package_ids = self.get_package_ids(latest_rrev)
```

Comment:
This doesn't look correct for production code, seems pretty arbitrary getting just the first package id in ``all_package_ids[0]``. Could be good as a simplification for testing, but looks dangerous here.